src/coreclr/jit/gentree.cpp

```diff
@@ -16487,7 +16487,7 @@ bool GenTree::IsBlockProfileUpdate()
 
     GenTree* const addr = lhs->AsIndir()->Addr();
 
-    return addr->IsIconHandle(GTF_ICON_BBC_PTR);
+    return addr->IsIntegralConst() && addr->IsIconHandle(GTF_ICON_BBC_PTR);
```

Comment:
I suppose the more technically correct test would be `IsCnsIntOrI` as `IsIntegralConst` will be `true` for `CNS_LNG` as well.